src/client/auth/scram.rs

```diff
@@ -304,18 +363,22 @@ fn parse_kvp(str: &str, expected_key: char) -> Result<String> {
 }
 
 /// Model of the first message sent by the client.
-struct ClientFirst {
+#[derive(Debug)]
+pub(crate) struct ClientFirst {
     source: String,
 
     message: String,
 
     gs2_header: Range<usize>,
 
     bare: Range<usize>,
+
+    nonce: String,
 }
 
 impl ClientFirst {
-    fn new(source: &str, username: &str, nonce: &str) -> Self {
+    fn new(source: &str, username: &str) -> Self {
+        let nonce = auth::generate_nonce();
```

Comment:
Since the nonce can't just be generated in the same scope where it's used like it was before, there isn't a strong need for the caller of this method to keep ownership of it, so to simplify things I moved the generation inside the method.